six/modules/c++/six.sicd/source/SICDMesh.cpp

```diff
@@ -0,0 +1,276 @@
+/* =========================================================================
+ * This file is part of six-c++
+ * =========================================================================
+ *
+ * (C) Copyright 2004 - 2014, MDA Information Systems LLC
+ *
+ * six-c++ is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; If not,
+ * see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include <six/sicd/SICDMesh.h>
+
+namespace
+{
+
+/*!
```

Comment:
Isn't all this serializer/deserializer pretty general purpose stuff that you'd want to reuse if you were implementing your own custom stuff into the SICD?  It seems like all of this should go in its own header.

Reply:
Agreed. I moved it to Serialize.h under the six module. I also provided a unittest for round-tripping various scalar and vector-of-scalar data types.